### PR TITLE
✨ Adds support for showing tooltip on hover, click events

### LINKED ISSUE #8: Add a demo for this appearing on a hover/click event

### PR DESCRIPTION
**Pull request by aendrew**
Resolves #8.

### CODE REVIEW

File: demos/src/hover.mustache [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	<div class='demo-tooltip-container' id="box-tooltip">
+
+		<button class='o-buttons o-buttons--big demo-tooltip-target' id="demo-tooltip-target">
+			Share
+		</div>
+
+		<div data-o-component="o-tooltip"
+		data-o-tooltip-position="right"
+		data-o-tooltip-target="demo-tooltip-target"
+		data-o-tooltip-show-on-hover="true">
```
Comment (JakeChampion) [inline demos/src/hover.mustache]: If this is not quoted I think it will return a boolean instead of a string, which may be useful when it comes to https://github.com/Financial-Times/o-tooltip/pull/10/files#diff-12b4ef0bf4145d4278dc13ca533b5197R75

Comment (aendrew) [inline demos/src/hover.mustache]: @JakeChampion AFAIK, `Element.getAttribute()` will always return a string (or `null`). I'd updated all the attributes in the documentation to reflect this fact (and the fact that quoting attributes is a good practice given that things like unquoted `mailto:` strings will break things. See [mothereff.in/unquoted-attributes](https://mothereff.in/unquoted-attributes) and [MDN: `Element.getAttribute()`](https://developer.mozilla.org/en/docs/Web/API/Element/getAttribute).

Comment (JakeChampion) [inline demos/src/hover.mustache]: Nice links, thanks :-)

I also checked with the dataset accessor approach and it also returns strings.

``` js
> document.body
> <body>​…​</body>​
> document.body.dataset.oToolTipShowOnHover = true
> true
> document.body
> <body data-o-tool-tip-show-on-hover=​"true">​…​</body>​
> document.body.dataset.oToolTipShowOnHover
> "true"
```

**Comment by alicebartlett** [comment]
This is great!

**Comment by aendrew** [comment]
@JakeChampion @alicebartlett I've added some unit tests, but have [disabled them](https://github.com/Financial-Times/o-tooltip/pull/10/files#diff-9e11dc14830621c20fe097c337e47ac6R96) because I cannot seem to get the events to fire. It may be how I'm stubbing `show`/`close` (if I change those from a stub to a spy the entire test suite blows up), or it may be some weird PhantomJS thing preventing me from firing synthetic mouse events, I'm really not sure. Any ideas very much welcome.

**Comment by aendrew** [comment]
Updated — tests now pass. :tada:

File: src/js/tooltip.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ class Tooltip {
 		this.visible = false;
 
 		this.delegates = {
-			target: new Delegate(this.target.targetEl),
+			target: new Delegate(document.getElementById(this.opts.target)),
```
Comment (JakeChampion) [inline src/js/tooltip.js]: Why was this changed? We already have a reference to the target element in `this.targetEl`.

Comment (aendrew) [inline src/js/tooltip.js]: @JakeChampion I didn't realise `Target` was being stubbed and so hard to restore it; tests were passing when I did a `document.getElementById` but not when I lacked it. Now realising that `Target` is stubbed, I restored it and have reverted to the previous code; see updated commit.

**Comment by aendrew** [comment]
Fixed merge conflicts.
